dev/com.ibm.ws.threading/src/com/ibm/ws/threading/internal/ThreadPoolController.java

```diff
@@ -565,7 +579,33 @@
             activeTask = new IntervalTask(this);
             timer.schedule(activeTask, interval, interval);
         }
+        /**
+         * if coreThreads has been configured to a small value, we will use the
+         * configured value as guidance for how large to make poolSize changes
+         */
+        if (coreThreads < NUMBER_CPUS * 2) {
+            poolChangeBasis = Math.max(1, coreThreads / 2);
```

Comment:
I am sorry to keep bringing up this particular logic.  Why is it < NUMBER_CPUS * 2 instead of < NUMBER_CPUS?  Why wouldn't it be just be poolChangeBasis = coreThreads?

Reply:
The default value for coreThreads is NUMBER_CPUS * 2  ... so the if clause checks whether coreThreads has been configured to a smaller value, overriding the default.
In the 18001 threadpool controller changes, a central part of the design was that it makes sense to adjust the default pool size by an increment based on NUMBER_CPUS, so that adjustments will increase/decrease the pool size by one thread per underlying h/w thread. Setting poolChangeBasis = coreThreads would double that, which is not the intent of these changes. 
Not sure if this answers your question ...

Reply:
It does make sense.  Should we have something like:

if (coreThreads < NUMBER_CPUS) {
    poolChangeBasis = Math.max(1, coreThreads);
} else if (coreThreads < NUMBER_CPUS * 2) {
....